pavelib/edraak.py

```diff
@@ -6,111 +6,219 @@
 
-    raise Exception('Undefined behaviour')
+            copyfile('django.po', latest)
+
+    run_for_js_nonjs(create_lastest)
+
+    sh('git checkout -- conf/')  # Undo brutal `i18n_tool generate` chenges
 
 
 @task
-def edraak_generate_files():
+def i18n_generate_latest():
     """
-    Append all Edraak strings to the original django.mo.
+    Compile localizable strings from sources, extracting strings first.
     """
-    lc_messages_dir = PLATFORM_ROOT / 'conf/locale/ar/LC_MESSAGES'
 
-    django_pofile_name = lc_messages_dir / 'django.po'
-    django_mofile_name = lc_messages_dir / 'django.mo'
+    cmd = 'i18n_tool generate'
+    sh(cmd + ' --rtl --strict')
 
-    djangojs_pofile_name = lc_messages_dir / 'djangojs.po'
-    djangojs_mofile_name = lc_messages_dir / 'djangojs.mo'
 
-    django_pofile = polib.pofile(django_pofile_name)
-    djangojs_pofile = polib.pofile(djangojs_pofile_name)
+@task
+@js_nonjs
+def i18n_make_missing(is_js, suffix):
+    with working_directory(ARABIC_LOCALE_DIR):
+        django = polib.pofile('django{}.po'.format(suffix))
+        django_latest = polib.pofile('latest-django{}.po'.format(suffix))
 
-    for resource in ('edraak-platform.po', 'edraak-platform-2015-theme.po'):
-        edraak_pofile_name = lc_messages_dir / resource
-        edraak_pofile = polib.pofile(edraak_pofile_name)
+        translated_msgids = {}
+        for entry in django.translated_entries():
+            translated_msgids[entry.msgid] = True
 
-        for entry in edraak_pofile:
-            django_pofile.append(entry)
+        for entry in reversed(django_latest):
+            if not entry.translated():
+                django_latest.remove(entry)
+            elif translated_msgids.get(entry.msgid):
+                django_latest.remove(entry)
 
-    for i, entry in enumerate(django_pofile):
-        django_pofile[i] = cleanup_po_entry_conflicts(entry)
+        print 'Added {} missing translations for django{}.po:'.format(
+            len(django_latest),
+            suffix,
+        )
 
-    # Save a backup in git for later inspection, and keep django.po untouched
-    customized_pofile_name = lc_messages_dir / 'django-edraak-customized.po'
-    django_pofile.save(customized_pofile_name)
-    django_pofile.save_as_mofile(django_mofile_name)
-    djangojs_pofile.save_as_mofile(djangojs_mofile_name)
+        django_latest.save('django{}-missing.po'.format(suffix))
+        path('latest-django{}.po'.format(suffix)).remove()
 
 
 @task
 @needs(
-    'pavelib.edraak.edraak_transifex_pull',
-    'pavelib.edraak.edraak_generate_files',
+    'pavelib.edraak.clean_repo_check',
+    'pavelib.i18n.i18n_clean',
+    'pavelib.edraak.i18n_transifex_pull_edx',
+    'pavelib.edraak.i18n_transifex_pull_edraak',
+    'pavelib.i18n.i18n_extract',
+    'pavelib.i18n.i18n_dummy',
+    'pavelib.edraak.i18n_generate_latest',
+    'pavelib.edraak.i18n_make_missing',
+    'pavelib.edraak.i18n_edraak_generate_files',
 )
 def edraak_i18n_pull():
```

Comment:
Why not naming it: `i18n_edraak_pull` instead? I think it would be more edxonic and straight forward. All of edX translations are managed by Transifex, so there's no need to specify it in translations commands to avoid understanding it as there's another organization dealing with translations together with Transifex.